src/main/kotlin/org/jitsi/jicofo/jibri/JibriSipGateway.kt

```diff
@@ -0,0 +1,162 @@
+            val jibriSession = JibriSession(
+                this,
+                conference.roomName,
+                iq.from,
+                config.pendingTimeout.seconds,
+                config.numRetries,
+                connection,
+                jibriDetector,
+                false,
+                iq.sipAddress,
+                iq.displayName, null, null, sessionId, null,
+                logger
+            )
+            sipSessions[iq.sipAddress] = jibriSession
+            try {
+                jibriSession.start()
+                logger.info("Started Jibri session")
+                JibriIq.createResult(iq, sessionId)
+            } catch (exc: StartException) {
+                val reason = exc.message
+                logger.warn("Failed to start a Jibri session: $reason", exc)
+                sipSessions.remove(iq.sipAddress)
+                when (exc) {
+                    is AllBusy -> error(iq, XMPPError.Condition.resource_constraint, "all Jibris are busy")
+                    is NotAvailable -> error(iq, XMPPError.Condition.service_unavailable, "no Jibri instances available")
+                    else -> error(iq, XMPPError.Condition.internal_server_error, reason)
+                }
+            }
+        } else {
+            // Bad request - no SIP address
+            error(iq, XMPPError.Condition.bad_request, "Stream ID is empty or undefined")
+        }
+
+    override fun onSessionStateChanged(
+        jibriSession: JibriSession,
+        newStatus: JibriIq.Status,
+        failureReason: JibriIq.FailureReason?
+    ) {
+        if (!sipSessions.containsValue(jibriSession)) {
+            logger.error("onSessionStateChanged for unknown session: $jibriSession")
+            return
+        }
+        publishJibriSipCallState(jibriSession, newStatus, failureReason)
+        if (JibriIq.Status.OFF == newStatus) {
+            val sipAddress = jibriSession.sipAddress
+            sipSessions.remove(sipAddress)
+            logger.info("Removing SIP call: $sipAddress")
+        }
+    }
+
+    /**
+     * Updates status of specific [JibriSession]. Jicofo adds multiple
+     * [SipCallState] MUC presence extensions to it's presence. One for
+     * each active SIP Jibri session.
+     * @param session the session for which the new status will be set
+     * @param newStatus the new status
+     * @param failureReason option error for OFF state
+     */
+    private fun publishJibriSipCallState(
+        session: JibriSession,
+        newStatus: JibriIq.Status,
+        failureReason: JibriIq.FailureReason?
+    ) {
+        val sipCallState = SipCallState().apply {
+            setState(newStatus)
+            this.failureReason = failureReason
+            sipAddress = session.sipAddress
+            sessionId = session.sessionId
+        }
+        logger.info("Publishing new state: ${session.sipAddress} ${sipCallState.toXML()} in: " + conference.roomName)
```

Comment:
Use string interpolation for `conference.roomName` too.

Reply:
Removed that, it's now part of the logger context 👍